src/tools/s2wasm.cpp

```diff
@@ -81,6 +83,24 @@ int main(int argc, const char *argv[]) {
            [&allowMemoryGrowth](Options *, const std::string &) {
              allowMemoryGrowth = true;
            })
+      .add("--emit-potential-traps", "",
```

Comment:
I get that this matches asm2wasm, but I feel like we should have made this flag something like `--trap-mode=<enum>`. Like, if Allow is the default, this flag is really a no-op. Maybe a followup fix?

Reply:
Well we don't have this hooked up in Emscripten yet, so now seems like a great time to make an asymmetric change.

Reply:
we should keep it matching asm2wasm, though? perhaps we should change both

Reply:
Yeah we can change both. But I agree that since we don't have the s2wasm one hooked up yet, that we might as well put what we finally want in s2wasm, then maybe fix asm2wasm in a followup PR.

Reply:
Well we'll need to make an Emscripten patch to change the plumbing for asm2wasm anyway, we can bundle that with the patch to add plumbing for s2wasm, and we can keep them the same in Binaryen in the meantime.

So:
Keep this as is for now
Follow up with a patch to Binaryen to change both sets of args to `--trap-mode=<blah>`
At the same time submit a patch to Emscripten to add s2wasm's invocation and change asm2wasm's.